Update readme (#14)
elan-ev/vosk-cli: README.md

# vosk-cli

![GitHub Workflow Status](https://img.shields.io/github/workflow/status/elan-ev/vosk-cli/Build%20and%20publish)
![GitHub](https://img.shields.io/github/license/elan-ev/vosk-cli)
![PyPI](https://img.shields.io/pypi/v/vosk-cli)

This python package serves as an Vosk interface for Opencast. It allows to generate subtitles (WebVTT files) from Video and Audio sources via Vosk.

## Installation

### 1. Install vosk-cli
To install the [latest stable version of vosk-cli](https://pypi.org/project/vosk-cli/), run

```
pip install vosk-cli
```

Alternatively, to install the latest development version, clone this project and inside the project directory run

```
pip install .
```

### 2. Install dependencies

vosk-cli uses ffmpeg to preprocess input files. The easiest way to install ffmpeg is by using a package manager.
To install ffmpeg on **Linux**, run

```

> sudo apt install ffmpeg

Reviewer: This wouldn't work for me. I'm on fedora and my default package manager is `dnf`. On Gentoo, it would be something else, on Arch, it would be something else again. It's basically impossible to list them all :D

That's why I would probably not try to list all options, but just tell users to “install FFmpeg” and potentially link http://ffmpeg.org/download.html

```
On **MacOS**, run 
```
brew install ffmpeg
```

If you are on **Windows** and/or want or need to install from source, visit

> [FFmpeg.org](https://ffmpeg.org/download.html) and follow the instructions for your operating system.

Reviewer: I don't think we want to support Windows, do we?

### 3. Download the language model

Go to [https://alphacephei.com/vosk/models](https://alphacephei.com/vosk/models) and download at least the English language model.

> You can unzip the folder of the language model into any directory, but it is recommended to create and use a `./models` folder in the project directory and to rename your unzipped model to `eng` for example. Please use 3 digit language codes for the directory name. The default and fallback language directory of vosk-cli is `eng`. The directory name should match the workflow operation field `language-code`.

Reviewer: ```suggestion
You can unzip the folder of the language model into any directory, but it is recommended to create and use a `./models` folder in the project directory.
```

- I would leave Opencast configuration out of this
- I would discourage people from naming models `eng`, since you don't know any longer, which model this is. There are several different modelf for English, for example.

### 4. Usage

> Now you are able to run `vosk-cli -i <input_file_path> -o <output_file_path> -m <3-digit-language-code>`.

Reviewer: The model does not have to be a `3-digit-language-code`. The language parameter would need that but that one is deprecated.

```suggestion
Now you are able to run `vosk-cli -i <input_file_path> -o <output_file_path> -m <model-name-or-path>`.
```

> It is also possible to use the absolute directory path of the language model instead of the 3 digit code. The output file path does not need to be an existing path, but can simply be any `name`. vosk-cli will then create a `name.vtt` file containing the transcribed text in your current directory.

Reviewer: ```suggestion
```

The name actually is a local path :D

> For example, if there is a `video.mp4` file in your download folder, you can run `vosk-cli -i Users/<username>/Downloads/video.mp4 -o text -m eng` (please replace `<username>` with your own username).

Reviewer: Should also work on macOS, I hope

```suggestion
For example, if there is a `video.mp4` file in your download folder, you can run `vosk-cli -i ~/Downloads/video.mp4 -o text -m eng`.
```
